src/execution_plan/execution_plan.h

```diff
@@ -17,14 +16,15 @@ typedef struct ExecutionPlan ExecutionPlan;
 struct ExecutionPlan {
 	OpBase *root;                       // Root operation of overall ExecutionPlan.
 	rax *record_map;                    // Mapping between identifiers and record indices.
-	ResultSet *result_set;              // ResultSet populated by this query.
 	QueryGraph *query_graph;            // QueryGraph representing all graph entities in this segment.
 	FT_FilterNode *filter_tree;         // FilterTree containing filters to be applied to this segment.
 	QueryGraph **connected_components;  // Array of all connected components in this segment.
 	// NOTE - segments and segment_count are only stored for proper freeing.
 	int segment_count;                  // Number of ExecutionPlan segments.
 	ExecutionPlan **segments;           // Partial execution plans scoped to a subset of operations.
 	ObjectPool *record_pool;
+	const char **returned_column_names; // The final result set column names.
+	bool is_union;                      // Indicates if the execution plan as a union of execution plans.
```

Comment:
This is currently always false, though that should be fixed by the change at 735. Do ExecutionPlans actually need to be aware of this at all, though? It looks like this is used as a switch for `optimizePlan`, but we can probably base the same switch on an AST check for a UNION clause - thoughts?

Reply:
removed